### PR TITLE
Feature/computer vision-Camera

### PR DESCRIPTION
**Pull request by MarlySaliba**
#43 #44 
Camera Terminé en attente de validation

### CODE REVIEW

**Review by fabricerenard12 — CHANGES_REQUESTED** [review]
Docstrings pour chaque méthode svp

File: src/camera.py
Comment (nguyenm31) [inline src/camera.py:1]: La classe caméra suivre un pattern de singleton

ex: 
```
class Singleton:
    _instance = None  # stockage de l’instance unique

    def __new__(cls, *args, **kwargs):
        if cls._instance is None:  # si aucune instance encore créée
            cls._instance = super().__new__(cls)
        return cls._instance

# Exemple d’utilisation
a = Singleton()
b = Singleton()
print(a is b)  # True → les deux variables pointent vers la même instance

```

File: src/camera.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.close()
+    
+
+if __name__ == "__main__":
+    print("Camera test starting")
+    cam = Camera()
+    cam2 = Camera()
+
+    print("WOww the singleton works bro!" if cam is cam2 else "What the helly...")
```
Comment (fabricerenard12) [inline src/camera.py:88]: 😭😭😭

File: src/cv_viewer/tracking_viewer.py [hunk 2]
```diff
@@ -32,13 +43,23 @@ def get_image_position(bounding_box_image, img_scale):
...
+        is_tracking_on (bool): Represents if the tracking is activated or not
+        label (int): The integer corresponding to the object
+    Returns:
+        None
+    """
     overlay = left_display.copy()
 
     line_thickness = 2
     for obj in objects.object_list:
         # filtre
```
Comment (fabricerenard12) [inline src/cv_viewer/tracking_viewer.py]: À retirer

File: src/cv_viewer/tracking_viewer.py [hunk 3]
```diff
@@ -32,13 +43,23 @@ def get_image_position(bounding_box_image, img_scale):
...
 
-        if obj.raw_label != label and label != -1: continue
+        if label is not None and obj.raw_label != label and label != -1: continue
```
Comment (nguyenm31) [inline src/cv_viewer/tracking_viewer.py]: what is the difference between label = None and label = -1 

(write meaning in doctsring if the same meaning we need to chose one and stick to it)

File: src/depth_sensing.py
Comment (nguyenm31) [inline src/depth_sensing.py]: let's add depth_sensing in another PR

File: src/main.py [hunk 4]
```diff
@@ -83,13 +83,15 @@ def main():
...
+    cv = ComputerVision(opt) 
+
     if opt.cv is not None:
         try:
             duration = float('inf') if opt.cv.lower() == 'inf' else float(opt.cv)
         except ValueError:
             raise ValueError(f"Invalid value for --cv: {opt.cv}")
         with torch.no_grad():
-            coordinated_target_list = detector.object_detection(duration, opt)
+            coordinated_target_list = cv.object_detection(duration, 39)
```
Comment (nguyenm31) [inline src/main.py]: why 39?

Comment (ydna985) [reply src/main.py]: 39 is the label for bottle. By the way, this file needs to be updated since computer vison was changed. We don't detect specific objects anymore.

**Review by nguyenm31 — APPROVED** [review]
